### ISSUE TITLE
SwiftLint ignore static func or class names

### ISSUE THREAD

**Issue by dimidrollko** [realm/SwiftLint#3127]
### New Issue Checklist

- [+] Updated SwiftLint to the latest version
- [+] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
I add custom rule to prevent usage of NSLocalizedString with magic strings. 
regex: "(NSLocalizedString\(\"\w+\")" that perfectly works in browser.
```
    NSLocalized_Magic_string:
      name: "Using NSLocalized string localization" # rule name. optional.
      regex: "NSLocalizedString" #Shortened version
      #regex: "(NSLocalizedString\\(\"\\w+\\\")" # matching pattern
      matching:
      message: "You should use localization keys instead"
      severity: error

```
```swift
// This triggers a violation:
NSLocalizedString("Test",....)
```
